jetty-http2/http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/Huffman.java

```diff
@@ -390,9 +394,20 @@ public static String decode(ByteBuffer buffer,int length)
         {
             int c = (current << (8 - bits)) & 0xFF;
             node = tree[node*256+c];
-            if (rowbits[node]==0 || rowbits[node] > bits) 
+
+            if (rowbits[node]==0 || rowbits[node] > bits)
+            {
+                int requiredPadding = 0;
+                for(int i=0; i<bits; i++)
+                    requiredPadding = (requiredPadding << 1) | 1;
+
+                if((c>>(8-bits)) != requiredPadding)
+                    throw new HpackException.StreamException("Incorrect padding");
+
                 break;
-            
+            }
+
+            // TODO why is this even here
             if (rowbits[node]==0)
                 throw new IllegalStateException();
```

Comment:
At the very least this also needs to be CompressionException now.

Reply:
this is unreachable code the if statement just above it already handles this case, so I don't know why it was there

Reply:
OK, then remove it!

Reply:
but the block above may not handle it?  if can fall through so it is reachable!   Let's walk through it in a hangout when you are ready